lnwallet/interface.go

```diff
@@ -636,6 +636,19 @@ func InternalKeyForAddr(wallet WalletController, netParams *chaincfg.Params,
 
 	walletAddr, err := wallet.AddressInfo(addr)
 	if err != nil {
+		// If the error is that the address can't be found, it is not
+		// an error. This happens when any channel which is not a custom
+		// taproot channel is cooperatively closed to an external P2TR
+		// address. In this case there is no internal key associated
+		// with the address. Callers can use the .Option() method to get
+		// an option value.
+		var managerErr waddrmgr.ManagerError
+		if errors.As(err, &managerErr) &&
+			managerErr.ErrorCode == waddrmgr.ErrAddressNotFound {
```

Comment:
I don't have the full pic here, but just wanna point out that there probably should have a dedicated method to do this job since this method is named `InternalKeyForAddr` and the caller would expect an error when there's no internal key. In addition, we are fixing the case "non-custom tap channel closed to an external addr", then what about "custom tap chan closed but no internal key found" case? Previously we'd have an error but now it's ignored. Is that handled elsewhere? cc @guggero

Reply:
We only need the internal key in case of a custom (tap) channel. So for a normal taproot channel, we don't want this to error out and return a `None`. If we also return a `None` for custom (tap) channels, then that is fine _here_, as it will lead to an error elsewhere.
So the fix is good, we basically shouldn't error out for anything that can happen during normal operation (and specifying an _external_ shutdown address definitely is normal operation).

Basically the
```
if walletAddr == nil {
		return none, nil
}
```

further below is what we thought would happen if a key wasn't found. But apparently an error is returned, so we need this special case.

Reply:
cool thanks for the explanation!